### PR TITLE
pd:  support dpa2

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
Support DPA-2 in paddle backend. This PR will be updated after #4414 is merged.

### Training curve:

training_curves_comparison_dpa2

### Accuracy test(left: paddle, right: torch):

image

Ralated optimization of Paddle framework:
- [x] https://github.com/PaddlePaddle/Paddle/pull/69349
- [x] https://github.com/PaddlePaddle/Paddle/pull/69333
- [x] https://github.com/PaddlePaddle/Paddle/pull/69479
- [x] https://github.com/PaddlePaddle/Paddle/pull/69515
- [x] https://github.com/PaddlePaddle/Paddle/pull/69487
- [x] https://github.com/PaddlePaddle/Paddle/pull/69661
- [x] https://github.com/PaddlePaddle/Paddle/pull/69660
- [x] https://github.com/PaddlePaddle/Paddle/pull/69596
- [x] https://github.com/PaddlePaddle/Paddle/pull/69556

## Summary by CodeRabbit

- **New Features**
	- Introduced new classes for molecular descriptors: `DescrptDPA2`, `DescrptBlockRepformers`, `DescrptSeTTebd`, and `DescrptBlockSeTTebd`.
	- Added new functions for tensor operations and descriptor management, enhancing the capabilities of the module.
	- Updated JSON configurations for multitask models to refine selection criteria and data paths.

- **Bug Fixes**
	- Improved error handling and parameter validation across various descriptor classes.

- **Documentation**
	- Enhanced test coverage for new descriptor functionalities and configurations.

- **Tests**
	- Added new test classes to validate the functionality of `DescrptDPA2` and multitask training scenarios.
	- Expanded test capabilities for descriptor classes based on installed dependencies.
	- Updated existing tests to support new configurations and functionalities.

### CODE REVIEW

**Comment by HydrogenSulfate** [comment]
@njzjz This PR is ready for review

File: deepmd/pd/model/descriptor/dpa2.py [hunk 1]
```diff
@@ -0,0 +1,897 @@
...
+        exclude_types: list[tuple[int, int]] = [],
+        env_protection: float = 0.0,
+        trainable: bool = True,
+        seed: Optional[Union[int, list[int]]] = None,
+        add_tebd_to_repinit_out: bool = False,
+        use_econf_tebd: bool = False,
+        use_tebd_bias: bool = False,
+        type_map: Optional[list[str]] = None,
+    ) -> None:
+        r"""The DPA-2 descriptor. see https://arxiv.org/abs/2312.15492.
```
Comment (iProzd) [inline deepmd/pd/model/descriptor/dpa2.py:99]: reference need to be updated after https://github.com/deepmodeling/deepmd-kit/pull/4483 is merged.

**Review by iProzd — APPROVED** [review]

File: deepmd/pd/model/descriptor/dpa2.py (hunk 1 above)
Comment (HydrogenSulfate) [reply deepmd/pd/model/descriptor/dpa2.py]: > reference need to be updated after #4483 is merged.

reference now updated in `dpa2.py`.

**Review by njzjz — APPROVED** [review]

**Comment by njzjz** [comment]
@coderabbitai resolve

**Comment by njzjz** [comment]
@coderabbitai resolve

File: deepmd/pd/utils/env.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+RESERVED_PRECISON_DICT = {
+    paddle.float16: "float16",
+    paddle.float32: "float32",
+    paddle.float64: "float64",
+    paddle.int32: "int32",
+    paddle.int64: "int64",
+    paddle.bfloat16: "bfloat16",
+    paddle.bool: "bool",
+}
+assert set(PRECISION_DICT.values()) == set(RESERVED_PRECISON_DICT.keys())
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/pd/utils/env.py]: @HydrogenSulfate This comment seems to be reasonable

[the remark this answers is not in the document]
Comment (HydrogenSulfate) [reply deepmd/pd/utils/env.py]: > ```diff
> RESERVED_PRECISON_DICT 
> ```

Oh this seems a typo that exist in other backends...<https://github.com/search?q=repo%3Adeepmodeling%2Fdeepmd-kit%20RESERVED_PRECISON_DICT%20&type=code>

[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/pd/utils/env.py]: Maybe I can revise it in a separated PR

[the remark this answers is not in the document]
Comment (HydrogenSulfate) [reply deepmd/pd/utils/env.py]: > Maybe I can revise it in a separated PR

typo in pd backend are fixed now.

**Comment by HydrogenSulfate** [comment]
@njzjz I noticed that there might be some subtle typos in the deepmd project. In Paddle, we have introduced a tool called typos in the pre-commit process, which can automatically detect potential typos. You can try it and added into deepmd if interested: <https://github.com/PaddlePaddle/Paddle/issues/69377>.

**Pull request merged**